pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -254,167 +254,152 @@
+             .mono {
+               font-family: 'Source Code Pro', monospace;
              }
              @media all and (prefers-color-scheme: dark) {
-               html, body {
-                 background-color: #333;
-                 color: #fff;
-               }
-               a, a:visited {
-                 color: #fff;
-               }
-               input {
-                 background: #333;
-                 color: #fff;
-                 border: 1px solid #7f7f7f;
-               }
-               input:focus {
-                 border: 1px solid #fff;
-               }
-             }
-             @media all and (min-width: 34.375rem) {
-               .tc-ns {
-                 text-align: center;
-               }
-               .pr0-ns {
-                 padding-right: 0;
-               }
-               .dib-ns {
-                 display: inline-block;
+               :root {
+                 --white: rgb(51, 51, 51);
+                 --black100: rgba(255,255,255,1);
+                 --black05: rgba(255,255,255,0.05);
+                 --black20: rgba(255,255,255,0.2);
                }
              }
              '''
     ==
     ;body
-      ;div#main
-        ;div#inner
-          ;p:"Urbit ID"
-          ;input(value "{(scow %p our)}", disabled "true", class "mono");
-          ;p:"Access Key"
-          ;p.f9.gray2
-            ; Get key from Bridge, or
-            ;span.mono.pr1:"+code"
-            ; in dojo
-          ==
-          ;form(action "/~/login", method "post", enctype "application/x-www-form-urlencoded")
-            ;input
-              =type  "password"
-              =name  "password"
-              =placeholder  "sampel-ticlyt-migfun-falmel"
-              =class  "mono"
-              =autofocus  "true";
-            ;input(type "hidden", name "redirect", value redirect-str);
-            ;button(type "submit"):"Continue"
-          ==
-        ==
-        ;footer.absolute.w-100
-          ;div.relative.w-100.tr.tc-ns
-            ;p.pr2.pr0-ns.pb2
-              ;a(href "https://bridge.urbit.org", target "_blank")
-                ;span.dn.dib-ns.pr1:"Open"
-                ; Bridge ↗
-              ==
-              ;a
-                =href  "https://urbit.org/using/install/#id"
-                =style  "margin-left: 8px; color: #2aa779;"
-                =target  "_blank"
-                ; Purchase
-                ;span.dn.dib-ns.pr1:"an Urbit ID"
-                ; ↗
-              ==
+      ;p:"Urbit ID"
+      ;input(value "{(scow %p our)}", disabled "true", class "mono");
+      ;p:"Access Key"
+      ;form(action "/~/login", method "post", enctype "application/x-www-form-urlencoded")
+        ;input
+          =type  "password"
+          =name  "password"
+          =placeholder  "sampel-ticlyt-migfun-falmel"
+          =class  "mono"
+          =required  "true"
+          =minlength  "27"
+          =maxlength  "27"
+          :: =pattern  "((\w{6})-?){4}"
```

Comment:
This can probably go?

Reply:
Well, I was hoping to do pattern matching on the code, but this fails parsing because of the () and {}

Reply:
You should be able to just `\{`?

Reply:
Alas, alack, I report a nack

Reply:
Have you tried: `"((\\w\{6})-?)\{4}";`?

```hoon
> ;input
  =pattern  "((\\w\{6})-?)\{4}";
[[%input [%pattern "((\\w\{6})-?)\{4}"] ~] ~]
```

Reply:
Can I humbly suggest `((?:[a-z]{6}-){3}(?:[a-z]{6}))` as the regex? It's a bit clunkier, but the existing one will match e.g. `sampelsampelsampelsampel` because the `-` is always optional. Gratuitous non-capture groups added for good measure (so we get a whole capture or nothing at all - not sure if that makes a difference for HTML validation anyway).

Also switched out `\w` for `[a-z]` because `\w` includes 0-9 which I imagine we don't want to consider valid in a code?